src/app/add-clients/pageComponents.tsx

```diff
@@ -0,0 +1,740 @@
+
+    div {
+        width: 100%;
+        margin: 0.15em 0;
+    }
+`;
+
+const Text = styled.p`
+    padding-bottom: 1em;
+`;
+
+const Subheading = styled.h2`
+    padding-bottom: 1em;
+`;
+
+const Heading = styled.h1`
+    padding-bottom: 1em;
+`;
+
+const Asterisk = styled.span`
+    color: ${(props) => props.theme.errorColor};
+    &:before {
+        content: "*";
+    }
+`;
+
+const StyledButton = styled.button`
+    text-align: center;
+    width: 100px;
+    height: 30px;
+    border-radius: 10px;
+    border: solid 1px ${(props) => props.theme.foregroundColor};
+    background-color: ${(props) => props.theme.primaryBackgroundColor};
+    color: ${(props) => props.theme.primaryForegroundColor};
+
+    &:hover {
+        background-color: ${(props) => props.theme.secondaryBackgroundColor};
+        color: ${(props) => props.theme.secondaryForegroundColor};
+    }
+`;
+
+const initialErrorMessages = {
+    nameErrorMessage: "N/A",
+    phoneErrorMessage: "",
+    addressErrorMessage: "N/A",
+    postcodeErrorMessage: "N/A",
+    numberAdultsErrorMessage: "N/A",
+    numberChildrenErrorMessage: "N/A",
+    nappyErrorMessage: "",
+};
+
+const RequestForm: React.FC = () => {
+    const [fullName, setFullName] = useState("");
+    const [phoneNumber, setPhoneNumber] = useState("");
+    const [addressLine1, setAddressLine1] = useState("");
+    const [addressLine2, setAddressLine2] = useState("");
+    const [addressTown, setAddressTown] = useState("");
+    const [addressCounty, setAddressCounty] = useState("");
+    const [addressPostcode, setAddressPostcode] = useState("");
+    const [numberAdults, setNumberAdults] = useState<GenderToQuantity>({
+        female: 0,
+        male: 0,
+        adult: 0,
+    });
+    const [numberChildren, setNumberChildren] = useState(0);
+    const [ageGenderChildren, setAgeGenderChildren] = useState<AgeGenderChild[]>([]);
+    const [dietaryRequirements, setDietaryRequirements] = useState<booleanGroup>({});
+    const [feminineProducts, setFeminineProducts] = useState({});
+    const [babyProducts, setBabyProducts] = useState<boolean | null>(null);
+    const [nappySize, setNappySize] = useState("");
+    const [petFood, setPetFood] = useState({});
+    const [otherItems, setOtherItems] = useState({});
+    const [deliveryInstructions, setDeliveryInstruction] = useState("");
+    const [extraInformation, setExtraInformation] = useState("");
+
+    const [errorMessages, setErrorMessages] = useState<ErrorMessages>(initialErrorMessages);
+
+    useEffect(() => {
+        const defaultAgeGenderChildren = Array.from(
+            { length: numberChildren },
+            (value, index): AgeGenderChild => {
+                return {
+                    key: index,
+                    gender: "child",
+                    age: null,
+                };
+            }
+        );
+        setAgeGenderChildren(defaultAgeGenderChildren);
+    }, [numberChildren]);
```

Comment:
instead of a `useEffect`, wouldn't it be more sensible to just have an if statement that checks whether `ageGenderChildren.length === numberChildren` and if not then add/remove elements from the array? That also means data isn't overwritten when the user changes `numberChildren`

Reply:
For some reason doing it this way doesn't overwrite the data when the user changes `numberChildren`. 

Thanks for pointing it out though. Will change it to an if-statement once we figure out why the current implementation is working.